torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1461,7 +1461,6 @@ def __getitem__(self, key: int) -> Any:
     def _reset_lazy_init(self) -> None:
         """
         Reset instance so :func:`_lazy_init` will run on the next forward.
-        Currently this is only called in __init__
```

Comment:
Removing this comment since it is not true. It is also called in `apply()`.

nit: I think as a general rule, we should avoid these kinds of comments since they are not maintainable. Someone who adds a call to `_reset_lazy_init()` somewhere may not know to update this comment.